### PR TITLE
fix(datasets): Cleanup of dependency

### PR DESCRIPTION
**Pull request by DimedS**
## Description
This pull request addresses issue #747.

### Modifications:
1. Added `dask-csvdataset`.
2. Removed `polars-genericdataset` as it currently doesn't exist.
3. Added `spark-sparkstreamingdataset`.
4. Restructured the `polars` and `spark` groups for consistency with other groups.
5. Removed `hdfs-base` and `s3fs-base` from `spark-sparkjdbcdataset` as they are not required. They were retained in other groups as they may be necessary depending on the data storage.
6. Updated `Jinja2` to `<3.2.0` for consistency with documentation requirements.
7. Removed the special line for `tables` on Windows, as it was relevant only for Python 3.8 and is no longer necessary.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/pyproject.toml [hunk 1]
```diff
@@ -248,8 +259,7 @@ test = [
     "scipy>=1.7.3",
     "packaging",
     "SQLAlchemy>=1.2",
-    "tables>=3.8.0; platform_system == 'Windows'",  # Import issues with python 3.8 with pytables pinning to 3.8.0 fixes this https://github.com/PyTables/PyTables/issues/933#issuecomment-1555917593
-    "tables~=3.6; platform_system != 'Windows'",
+    "tables~=3.6",
```
Comment (ankatiyar) [inline kedro-datasets/pyproject.toml:262]: Does this work with a higher pin?

Comment (astrojuanlu) [reply kedro-datasets/pyproject.toml:262]: Yeah please, would be good to try `tables>=3.6`

File: kedro-datasets/pyproject.toml [hunk 2]
```diff
@@ -140,8 +144,15 @@ snowflake = ["kedro-datasets[snowflake-snowparktabledataset]"]
 spark-deltatabledataset = ["kedro-datasets[spark-base,hdfs-base,s3fs-base,delta-base]"]
 spark-sparkdataset = ["kedro-datasets[spark-base,hdfs-base,s3fs-base]"]
 spark-sparkhivedataset = ["kedro-datasets[spark-base,hdfs-base,s3fs-base]"]
-spark-sparkjdbcdataset = ["kedro-datasets[spark-base,hdfs-base,s3fs-base]"]
-spark = ["kedro-datasets[spark-deltatabledataset]"]
+spark-sparkjdbcdataset = ["kedro-datasets[spark-base]"]
```
[the remark this answers is not in the document]
Comment (DimedS) [reply kedro-datasets/pyproject.toml:147]: I believe JDBC is intended for relational databases and cannot be used with HDFS or S3FS?

**Review by ankatiyar — APPROVED** [review]
LGTM!

**Review by astrojuanlu — APPROVED** [review]
🚀

**Pull request merged**